alibi/explainers/backends/cfrl_tabular.py

```diff
@@ -256,25 +255,32 @@ def generate_condition(X_ohe: np.ndarray,
 
     Returns
     -------
-    Conditional vector.
+        Conditional vector.
     """
+    # Define conditional vector buffer
+    C = []
+
     # Generate numerical condition vector.
-    C_num = generate_numerical_condition(X_ohe=X_ohe,
-                                         feature_names=feature_names,
-                                         category_map=category_map,
-                                         ranges=ranges,
-                                         immutable_features=immutable_features,
-                                         conditional=conditional)
+    if len(feature_names) > len(category_map):
+        C_num = generate_numerical_condition(X_ohe=X_ohe,
+                                             feature_names=feature_names,
+                                             category_map=category_map,
+                                             ranges=ranges,
+                                             immutable_features=immutable_features,
+                                             conditional=conditional)
+        C.append(C_num)
 
     # Generate categorical condition vector.
-    C_cat = generate_categorical_condition(X_ohe=X_ohe,
-                                           feature_names=feature_names,
-                                           category_map=category_map,
-                                           immutable_features=immutable_features,
-                                           conditional=conditional)
+    if len(category_map):
+        C_cat = generate_categorical_condition(X_ohe=X_ohe,
+                                               feature_names=feature_names,
+                                               category_map=category_map,
+                                               immutable_features=immutable_features,
+                                               conditional=conditional)
+        C.append(C_cat)
 
     # Concatenate numerical and categorical conditional vectors.
-    return np.concatenate([C_num, C_cat], axis=1)
+    return np.concatenate(C, axis=1)
```

Comment:
Not sure I understand the reason for this change... Have the underlying `generate...` functions changed too?

Reply:
Also it seems that introducing `C=[]` and then appending to it twice just introduces unnecessary lines of code compared to the older version or am I missing something?

Reply:
To cover the cases where the dataset contains only numerical or only categorical

Reply:
No, the `generate...` function did not change